sys/aarch64/trap.c

```diff
@@ -123,7 +125,7 @@ void user_trap_handler(mcontext_t *uctx) {
       break;
 
     case EXCP_SVC64:
-      syscall_handler(ESR_ELx_SYSCALL(esr), ctx);
+      syscall_handler(ESR_ELx_SYSCALL(esr), ctx, &result);
```

Comment:
Maybe it's a good idea instead using `ESR_ELx_EXCEPTION(esr)`, use `exc_code` like you have done in switch above.

Reply:
This is line uses `ESR_ELx_SYSCALL`, not `ESR_ELx_EXCEPTION`.

Reply:
My bad.